client/src/main/java/org/apache/rocketmq/client/impl/consumer/ConsumeMessageConcurrentlyService.java

```diff
@@ -190,7 +201,7 @@ public void submitConsumeRequest(
             try {
                 this.consumeExecutor.submit(consumeRequest);
             } catch (RejectedExecutionException e) {
-                this.submitConsumeRequestLater(consumeRequest);
+                this.submitConsumeRequestLater(consumeRequest, DEFAULT_CONSUME_REQUEST_LATER_DELAY_MILLS);
```

Comment:
After the consumeRequestQueue is full, it will still be placed in thread pool of scheduledExecutorService, which I don't think is of much significance.

Reply:
Yeah, you are right, just moving the task from LinkedBlockingQueue to DelayedWorkQueue (binary heap based).